cmd/skopeo/signing.go

```diff
@@ -88,3 +89,137 @@ var standaloneVerifyCmd = cli.Command{
 	ArgsUsage: "MANIFEST DOCKER-REFERENCE KEY-FINGERPRINT SIGNATURE",
 	Action:    standaloneVerify,
 }
+
+func addSignature(context *cli.Context) error {
+	if len(context.Args()) != 2 {
+		return errors.New("Usage: skopeo add-signature reference key-fingerprint")
+	}
+	ref := context.Args()[0]
+	fingerprint := context.Args()[1]
+
+	imageRef, err := transports.ParseImageName(ref)
+	if err != nil {
+		return fmt.Errorf("Error parsing %s: %v", ref, err)
+	}
+	name := transports.ImageName(imageRef)
+
+	// TODO: the signed named reference is always relative to the transport.  Is this the right value?
+	signReference := imageRef.StringWithinTransport()
```

Comment:
Okay, sounds simple enough - default to DockerReference().String() if possible, overridden with a CLI flag, and if neither is possible, return an error.  Fixing.